### ISSUE TITLE
INSTALL.pl attempts to connect to https service on :21

### ISSUE THREAD

**Issue by pdl** [Ensembl/ensembl-vep#1302]
## Describe the issue

I am attempting to get the ensembl cache for 108 with INSTALL.pl, but there have been changes to how URLs are retrieved which cause the process to either hang or eventually fail.

It looks like the URLs being retrieved in `download_to_file` are:

    https://ftp.ensembl.org:21/pub/release-108/variation/indexed_vep_cache/homo_sapiens_vep_108_GRCh38.tar.gz

Note that this is passed into curl, which tries to make HTTPS requests to :21, whereas in fact the FTP service runs on :21; curl retrieves the URL fine without the :21 which is added via a regex in a preceding line.

### System
- VEP version: 108.1
- VEP Cache version: 108.1
- Perl version: As per image
- OS: Singularity, built from Docker image running on 18.04
- tabix installed Yes, in Docker image

### Full VEP command line

I've used the perl debugger to go find where the issue seems to be:

```
perl -d /opt/vep/src/ensembl-vep/INSTALL.pl -a cfp -s homo_sapiens -y GRCh38 -g dbNSFP,CADD,SpliceAI
```

### Full error message

```
  DB<15> n
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1768):
1768:     $url =~ s/([a-z])\//$1\:21\// if $url =~ /ftp/ && $url !~ /\:21/;
  DB<15> n
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1770):
1770:     if($CAN_USE_CURL) {
  DB<15> print $url
https://ftp.ensembl.org:21/pub/release-108/variation/indexed_vep_cache/homo_sapiens_vep_108_GRCh38.tar.gz
  DB<16> n
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1771):
1771:       my $response = `curl -s -o $file -w '%{http_code}' --location "$url" `;
  DB<16> n
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1772):
1772:       if ( $response != 200 && $response != 226) {
  DB<16> print $response
000
  DB<17> n
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1773):
1773:         print "curl failed ($response), trying to fetch using LWP::Simple\n" unless $QUIET;
  DB<17> n
curl failed (000), trying to fetch using LWP::Simple
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1774):
1774:         $CAN_USE_CURL = 0;
  DB<17> n
main::download_to_file(/opt/vep/src/ensembl-vep/INSTALL.pl:1775):
1775:         download_to_file($url, $file);
  DB<17> n
LWP::Simple failed (501), trying to fetch using HTTP::Tiny
ERROR: Failed last resort of using HTTP::Tiny to download https://ftp.ensembl.org:21/pub/release-108/variation/indexed_vep_cache/homo_sapiens_vep_108_GRCh38.tar.gz
 at /opt/vep/src/ensembl-vep/INSTALL.pl line 1816.
        main::download_to_file("https://ftp.ensembl.org:21/pub/release-108/variation/indexed_"..., "/www/dp13/decipher-vep/vep/108.1/.vep/tmp/homo_sapiens_vep_10"...) called at /opt/vep/src/ensembl-vep/INSTALL.pl line 1792
        main::download_to_file("https://ftp.ensembl.org:21/pub/release-108/variation/indexed_"..., "/www/dp13/decipher-vep/vep/108.1/.vep/tmp/homo_sapiens_vep_10"...) called at /opt/vep/src/ensembl-vep/INSTALL.pl line 1775
        main::download_to_file("https://ftp.ensembl.org/pub/release-108/variation/indexed_vep"..., "/www/dp13/decipher-vep/vep/108.1/.vep/tmp/homo_sapiens_vep_10"...) called at /opt/vep/src/ensembl-vep/INSTALL.pl line 1277
        main::cache() called at /opt/vep/src/ensembl-vep/INSTALL.pl line 267
Debugged program terminated.  Use q to quit or R to restart,
use o inhibit_exit to avoid stopping after program termination,
h q, h R or h o to get additional info.
```

**Comment by pdl** [comment]
Related work by @nuno-agostinho

- https://github.com/Ensembl/ensembl-vep/pull/1286
- https://github.com/Ensembl/ensembl-vep/pull/1284

**Comment by pdl** [comment]
[The regex which is inserting the port number](https://github.com/Ensembl/ensembl-vep/blob/release/108/INSTALL.pl#L1768)

```
  $url =~ s/([a-z])\//$1\:21\// if $url =~ /ftp/ && $url !~ /\:21/;
```

The url matches ftp, even if it begins `https://ftp.ensembl.org...`
should this read `if $url =~ /^ftp:/ && $url !~ /\:21/;`?

**Comment by nuno-agostinho** [comment]
Hey @pdl, hope you are having a great day!

Thank you for your detailed report and narrowing down the issue. We intend to fix this as soon as possible.

In the meantime, you can change the URLs used to fetch cache and FASTA like so:
```
perl INSTALL.pl --CACHEURL ftp://ftp.ensembl.org/pub/release-108/variation/vep --FASTAURL ftp://ftp.ensembl.org/pub/release-108/fasta/
```

Please feel free to reply back if you find any other issues. Thanks again!

Kind regards,
Nuno

**Comment by pdl** [comment]
@nuno-agostinho Thanks - in case it's relevant for the fix and for others, that doesn't work for me as it fails `is_url` on https://github.com/Ensembl/ensembl-vep/blob/release/108/INSTALL.pl#L1274 and therefore jumps to the copy, which fails as it's not a filesystem path.

**Comment by nuno-agostinho** [comment]
Hi @pdl,

We released [VEP 108.2](https://github.com/Ensembl/ensembl-vep/releases/tag/release%2F108.2) to fix the issues reported in this thread.

Thanks for reporting them and sorry for the inconvenience!

Best regards,
Nuno
